pandas/errors/__init__.py

```diff
@@ -208,3 +208,9 @@ class InvalidIndexError(Exception):
 
     .. versionadded:: 1.1.0
     """
+
+
+class MismatchedIndexValueError(ValueError):
```

Comment:
I would just leave this as a ValueError and not make a new one.

Reply:
I can take it back to ValueError. It very much upsets me to not have specific errors, but it's your library.

Reply:
`IncorrectValuesError` any better? (I am making your changes and waiting for suite to pass)

Reply:
This might be a stupid question @jreback but do I just edit the restructured text document or is there some tooling I need to appreciate?